### ISSUE TITLE
Disable registration

### ISSUE THREAD

**Issue by andrewvaughan** [gramps-project/Gramps.js#338]
Apologies if this is already a feature and I'm just somehow  missing it - my family hopes to have a private (but publicly available) Gramps installation. This would mean disabling the Registration form in the login page. Is this possible? If not, please consider it as a feature, as I imagine this is a significant use-case.

**Comment by DavidMStraub** [comment]
Hi, you could hide the link using `hideRegisterLink` (see [here](https://www.grampsweb.org/frontend-config/)) from the login page. This would techncally still allow registrations via the API though. But registered users are disabled until the tree owner manually enables them, so I'm not sure why anyone not entitled to access should do it.

**Comment by zeitchef** [comment]
Not sure about OP but I've installed via Docker and I would really love a environment variable to disable the registration link - would something like that be feasible?

**Comment by andrewvaughan** [comment]
> Not sure about OP but I've installed via Docker and I would really love a environment variable to disable the registration link - would something like that be feasible?

You could extend the `Dockerfile` to copy the `config.js` file in with the configuration. Thats what I ended up doing. I definitely would like to see it completely disable registration, though, as this is a portal to a LOT of PII.

Thanks all!

**Comment by zeitchef** [comment]
I'm using Docker Compose and I was able to copy a `config.js` into the container with the following:

```
volumes:
  - '/path/to/data/config.js:/app/static/config.js'
```

but this is having no effect. I've confirmed the file is copied correctly and has the following config:

```
window.grampsjsConfig = {
  hideRegisterLink: true
}
```

Am I missing something?

**Comment by DavidMStraub** [comment]
> as this is a portal to a LOT of PII

Can you please elaborate? Would like to understand better.

**Comment by DavidMStraub** [comment]
I'm still waiting for a rationale why registrations via the API (but disabled in the frontend) are an issue, given that unconfirmed users have no rights. In any case, since this is a backend issue, I think this one here can be closed.

**Comment by andrewvaughan** [comment]
> I'm still waiting for a rationale why registrations via the API (but disabled in the frontend) are an issue, given that unconfirmed users have no rights. In any case, since this is a backend issue, I think this one here can be closed.

When you say "Unconfirmed users have no rights" does that mean they have no access to any PII? Maybe that's where I'm missing.

In regards to closing - it still seems like this is an inconsistency or vector for attack if someone wanted to flood account registration and make an admin nightmare. Maybe just make it a parking lot item for later?

**Comment by DavidMStraub** [comment]
> When you say "Unconfirmed users have no rights" does that mean they have no access to any PII? Maybe that's where I'm missing.

Yes. Authorization works via JSON web tokens, and a disabled user cannot obtain one, so cannot access *anything* else than the public internet.

**Comment by lucaberti** [comment]
> I'm still waiting for a rationale why registrations via the API (but disabled in the frontend) are an issue, given that unconfirmed users have no rights. In any case, since this is a backend issue, I think this one here can be closed.

Hello, since I asked to remove registration in another post, let me post here my take on this question. Lest's do ite with example: a while ago I created a new wordpress site, forgetting to disable user registration. When I noticed that - some 2 hours later - I already had 5 new user to accept. No harm done, since WP does it the same way: users don't have permission if not accepted. But, still, they populated my DB and I had to remove them. If the 2 hours-5 scamuser is the initial ratio, I'm quite concerned on how many scam-user I would have to refuse and how often I have to maitain the Gramps installation. 
This is my reason for asking a way to disable the signup feature alltoghether.
Thank you

**Comment by DavidMStraub** [comment]
Hi,
ok got it. I never got any scam user on Gramps Web but then again it's not as popular as Wordpress :laughing: 

I think it's fine to add a new feature to Web API if someone is willing to contribute it. Shifting the issue to the appropriate repo.
